### PR TITLE
Add Swedish NER Corpus

### PR DESCRIPTION
**Pull request by abhishekkrthakur**

### CODE REVIEW

File: datasets/swedish_ner_corpus/swedish_ner_corpus.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+                        sentence_counter,
+                        {
+                            "id": str(sentence_counter),
+                            "words": current_words,
+                            "labels": current_labels,
+                        },
+                    )
+                    sentence_counter += 1
+                    current_words = []
+                    current_labels = []
```
Comment (patrickvonplaten) [inline datasets/swedish_ner_corpus/swedish_ner_corpus.py:67]: I'd set a breakpoint here, just to be sure the code actually reaches `yield`

File: datasets/swedish_ner_corpus/swedish_ner_corpus.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+                row = row.rstrip()
+                row_split = row.split("\t")
+                if len(row_split) > 1:
+                    token, label = row_split
+                    current_words.append(token)
+                    current_labels.append(label)
+                else:
+                    if not current_words:
+                        continue
+                    assert len(current_words) == len(current_labels)
```
Comment (patrickvonplaten) [inline datasets/swedish_ner_corpus/swedish_ner_corpus.py:56]: assert message would be nice

File: datasets/swedish_ner_corpus/swedish_ner_corpus.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+    def _generate_examples(self, datapath):
+        sentence_counter = 0
+        with open(datapath, encoding="utf-8") as f:
+            current_words = []
+            current_labels = []
+            for row in f:
+                row = row.rstrip()
+                row_split = row.split("\t")
+                if len(row_split) > 1:
```
Comment (patrickvonplaten) [inline datasets/swedish_ner_corpus/swedish_ner_corpus.py:49]: Think it's better to replace this with `if len(row_split) == 2` and then add a nice error message if `if len(row_split)` has an unexpected length

File: datasets/swedish_ner_corpus/swedish_ner_corpus.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+_DESCRIPTION = """\
+Webbnyheter 2012 from Spraakbanken, semi-manually annotated and adapted for CoreNLP Swedish NER. Semi-manually defined in this case as: Bootstrapped from Swedish Gazetters then manually correcte/reviewed by two independent native speaking swedish annotators. No annotator agreement calculated.
+"""
+_HOMEPAGE_URL = "https://github.com/klintan/swedish-ner-corpus"
+_TRAIN_URL = "https://raw.githubusercontent.com/klintan/swedish-ner-corpus/master/train_corpus.txt"
+_TEST_URL = "https://raw.githubusercontent.com/klintan/swedish-ner-corpus/master/test_corpus.txt"
+
+
+class SwedishNERCorpusDataset(datasets.GeneratorBasedBuilder):
```
Comment (lhoestq) [inline datasets/swedish_ner_corpus/swedish_ner_corpus.py:28]: Could you rename it `SwedishNERCorpus` ?
The dataset class must be the camel case equivalent of `swedish_ner_corpus` which is the script name.

File: datasets/swedish_ner_corpus/swedish_ner_corpus.py [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "id": datasets.Value("string"),
+                    "tokens": datasets.Sequence(datasets.Value("string")),
+                    "ner_tags": datasets.Sequence(
+                        datasets.features.ClassLabel(names=["0", "LOC", "MISC", "ORG", "PER"])
+                    ),
```
Comment (lhoestq) [inline datasets/swedish_ner_corpus/swedish_ner_corpus.py:40]: perfect !

File: datasets/swedish_ner_corpus/swedish_ner_corpus.py [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Lint as: python3
+import datasets
+
+
```
Comment (lhoestq) [inline datasets/swedish_ner_corpus/swedish_ner_corpus.py:19]: Could you add a `_CITATION` as well ?

**Review by lhoestq — COMMENTED** [review]
Really cool ! Thanks for adding it :)

Could you also add a dataset card ? see the template guide https://github.com/huggingface/datasets/blob/master/templates/README_guide.md

**Review by patrickvonplaten — APPROVED** [review]
LGTM! @lhoestq - is it still ok if datasets have no config?

**Comment by lhoestq** [comment]
Yes the use of configs is optional

File: datasets/swedish_ner_corpus/swedish_ner_corpus.py (hunk 6 above)
Comment (abhishekkrthakur) [reply datasets/swedish_ner_corpus/swedish_ner_corpus.py:19]: there is none for this. so ill add none.

Comment (lhoestq) [reply datasets/swedish_ner_corpus/swedish_ner_corpus.py:19]: Oh I didn't know. Leave it that way then

**Comment by yjernite** [comment]
@abhishekkrthakur we want to keep track of the information that is and isn't in the dataset cards so we're asking everyone to use the full template :) If there is some information in there that you really can't find or don't feel qualified to add, you can just leave the `[More Information Needed]` text

**Pull request merged**